### ISSUE TITLE
Possible to expose prop to delay date parsing/formatting to onBlur?

### ISSUE THREAD

**Issue by roginfarrer** [gpbl/react-day-picker#631]
Sandbox: https://codesandbox.io/s/zkn6oqnq4l

I'm having some trouble using `dateFns` for the `formatDate` and `parseDate`. (I recognize that this library only supports Moment currently, but bare with me.) When using a custom `format` and `dateFns.format` or `dateFns.parse`, the input is impossible to edit. See a similar issue with Moment here: #584. As far as I know, `dateFns` does not offer a "forgiving" mode.

I think this issue could be addressed by exposing an option to postpone the execution of the parse/format functions until the `onBlur` event. That way the user has free range to edit the input, then the string can be formatted once the user is finished.

I'm open to other solutions, but I thought this could work.

**Comment by gpbl** [comment]
This is a legit improvement (beside the issues you are having which I'll read later :))

**Comment by roginfarrer** [comment]
@gpbl I put together a fork that accomplishes what I laid out above. Doesn't flip it as a prop yet, but I wanted to get some thoughts on it.

[See diff](https://github.com/gpbl/react-day-picker/compare/master...roginfarrer:master#diff-e2a925e665d836a09575f79095191746)

The big change is that `inputValue` is maintained in state separate from value. There is now also a check to make sure that the user's input is parseable. If it is, great, proceed and update the state. If not, revert to the previous `inputValue`.

**Comment by gpbl** [comment]
I don't think all these is needed:

> I'm having some trouble using dateFns for the formatDate and parseDate.

See this example: 
https://codesandbox.io/s/r44xwwj35o

Hope it helps!
